modules/pytket-qiskit/pytket/extensions/qiskit/backends/ibm.py

```diff
@@ -340,6 +307,57 @@ def characterisation(self) -> Dict[str, Any]:
     def backend_info(self) -> BackendInfo:
         return self._backend_info
 
+    @classmethod
+    def _get_backend_info(cls, backend: "_QiskIBMQBackend") -> BackendInfo:
+        config = backend.configuration()
+        characterisation = process_characterisation(backend)
+        averaged_errors = get_avg_characterisation(characterisation)
+        characterisation_keys = [
+            "t1times",
+            "t2times",
+            "Frequencies",
+            "GateTimes",
+        ]
+        arch = characterisation["Architecture"]
+        # filter entries to keep
+        filtered_characterisation = {
+            k: v for k, v in characterisation.items() if k in characterisation_keys
+        }
+        supports_mid_measure = config.simulator or config.multi_meas_enabled
+        supports_fast_feedforward = supports_mid_measure
+        # simulator i.e. "ibmq_qasm_simulator" does not have `supported_instructions`
+        # attribute
+        gate_set = _tk_gate_set(backend)
+        backend_info = BackendInfo(
+            cls.__name__,
+            backend.name(),
+            __extension_version__,
+            arch,
+            gate_set,
+            supports_midcircuit_measurement=supports_mid_measure,
+            supports_fast_feedforward=supports_fast_feedforward,
+            all_node_gate_errors=characterisation["NodeErrors"],
+            all_edge_gate_errors=characterisation["EdgeErrors"],
+            all_readout_errors=characterisation["ReadoutErrors"],
+            averaged_node_gate_errors=averaged_errors["node_errors"],
+            averaged_edge_gate_errors=averaged_errors["edge_errors"],
+            averaged_readout_errors=averaged_errors["readout_errors"],
+            misc={"characterisation": filtered_characterisation},
+        )
+        return backend_info
+
+    @classmethod
+    def available_devices(cls, **kwargs: KwargTypes) -> List[BackendInfo]:
+        hub = kwargs.get("hub")
+        group = kwargs.get("group")
+        project = kwargs.get("project")
+        IBMQ.load_account()
+        provider = IBMQ.get_provider(hub=hub, group=group, project=project)
```

Comment:
I suggest turning _get_provider in to a static function which takes an optional QiskitConfig instance, and use that here
this method should take an optional provider (like __init__) and use that if provided.